### DISCUSSION TITLE
New documents are automatically tagged and assigned a type, why?

### DISCUSSION THREAD

**Discussion by wsw70** [paperless-ngx/paperless-ngx#7461] [Support] [answered]
Whenever I add a document, it gets tagged with the same tag (`carte d'identité` in my case) and assigned to a type (`identité` in my case). This may be some kind of auto-tagging (that I saw being mentioned), but I cannot find a way to either influence it, or reduce its sensitivity. The logs show lines such as the one below when adding a document

```
[2024-08-13 10:46:52,215] [INFO] [paperless.handlers] Assigning document type identité to 2024-08-13 Attestations d'assurance Responsabilité Civile scolaire
[2024-08-13 10:46:52,231] [INFO] [paperless.handlers] Tagging "2024-08-13 Attestations d'assurance Responsabilité Civile scolaire" with "carte d'identité"
```

How should I approach this problem? (I am just starting to use paperless-ng so everything is fresh and not tuned yet)

**Comment by Criomby** [comment, marked as answer]
You probably have the tag and type you created set to auto-assign, which means the algorithm will set the tag/type based on its training data, i.e. your documents.
If you are just starting to add documents the algorithm might not have a sufficient set of docs available to accurately assign tags/types. This will improve as you add more documents to your database.

If you do not want the algorithm to automatically assign the specific tag/type, you can change it to no be assigned automatically or assign the tag/type based on some pattern it finds in the document content (e.g. assign tag "Company X" to documents containing the string "Company X").
You will have to do this on a per tag/type basis and can set the behaviour in the tag/type (sidebar) menu when you click on the tag/type you want to change.
The same applies to other stuff like correspondents.

**Reply by wsw70** [reply]
> You probably have the tag and type you created set to auto-assign, which means the algorithm will set the tag/type based on its training data, i.e. your documents. If you are just starting to add documents the algorithm might not have a sufficient set of docs available to accurately assign tags/types. This will improve as you add more documents to your database.

Yes, it was that, thank you. I had tags and documents set to the default of "Auto".

I wonder, though, how the training is supposed to work then: I dumped a bunch of documents, all got tagged with one tag (despite having other tags also set to "Auto") - so am I supposed to untag them and Paperless-ng will learn that I removed them and make something out of it? (like not tagging with that tag some documents? that have some content?) → I was expecting the other way round = start automatically tagging when it looks like it learned which documents are manually tagged with a tag.

> If you do not want the algorithm to automatically assign the specific tag/type, you can change it

I will try this, thanks.

**Reply by Criomby** [reply]
I suggest you take a look at the docs:

https://docs.paperless-ngx.com/advanced_usage/#matching
https://docs.paperless-ngx.com/administration/#retagger

**Reply by wsw70** [reply]
> https://docs.paperless-ngx.com/advanced_usage/#matching 

I read that in the meantime (notably https://docs.paperless-ngx.com/advanced_usage/#automatic-matching) but it is not clear to me whether the algorithm works in the background *regardless of settings for tags*, or whether tag(s) have to have "Auto" set for the neural network to learn from them. The former seems more logical (paperless learns for some time and then, when I set tag(s) to Auto they magically know where to be applied because of the "background" learning done so far).

**Reply by shamoon** [reply]
Yes. You can also just leave things on auto and “teach” it ie correct the assignments and yes they will improve after you’ve processed a few docs (and continue to improve)

**Reply by wsw70** [reply]
> Yes. You can also just leave things on auto and “teach” it ie correct the assignments and yes they will improve after you’ve processed a few docs (and continue to improve)

@shamoon I am not sure I understand (sorry for that, I am not a native speaker of English)
Do you mean that
- I can leave the setting to "None" (or predictably automated, such as via strings) and the algorithm will learn in the background on the tags/documents I set for imported data → and therefore I can at some point switch to "Auto" and hope for the best given that the NN has learned in the meantime
- or I can leave at "Auto" but have possibly chaotic tagging until the time the NN learns.
- or any of these

If this is the third point (any approach will result in a trained NN), I am not sure what would be the advantage to leaving it on Auto?

**Reply by shamoon** [reply]
Any of them are fine. The 'advantage' of leaving them on auto is that you don't have to change them later =)

Most people tend to review new documents as they're added so perhaps it's easier to correct the first few and eventually you'll find you need to make fewer or even no corrections (but algorithms are never perfect).

So really, whatever you prefer

**Reply by wsw70** [reply]
> So really, whatever you prefer

Thank you - it is all clear now

**Comment by github-actions** [comment]
This discussion has been automatically closed because it was marked as answered. Please see our [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/dev/CONTRIBUTING.md#automatic-repository-maintenance) for more details.

**Comment by github-actions** [comment]
This discussion has been automatically locked since there has not been any recent activity after it was closed. Please open a new discussion for related concerns. See our [contributing guidelines](https://github.com/paperless-ngx/paperless-ngx/blob/dev/CONTRIBUTING.md#automatic-repository-maintenance) for more details.
